### PR TITLE
Update android implementation

### PR DESCRIPTION
**Pull request by damieng**
This fixes or solves the following issues relating to Android support:

1. Logout URL defaults to app so it work by default #45 
2. Provides support for Custom Chrome Tabs #21 
3. New optional Auth0ClientActivity base class to auto-wire up needed events
4. Cleaned up Android test project

### CODE REVIEW

File: src/Auth0.OidcClient.Android/AndroidWebViewBase.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+            if (string.IsNullOrWhiteSpace(options.EndUrl))
+                throw new ArgumentException("Missing EndUrl", nameof(options));
+
+            var tcs = new TaskCompletionSource<BrowserResult>();
+
+            void Callback(string response)
+            {
+                ActivityMediator.Instance.ActivityMessageReceived -= Callback;
+
+                var cancelled = response == "UserCancel";
```
Comment (lbalmaceda) [inline src/Auth0.OidcClient.Android/AndroidWebViewBase.cs]: should this be part of a constant? here you're comparing, but where is it set originally?

File: src/Auth0.OidcClient.Android/Auth0ClientActivity.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        protected override void OnResume()
+        {
+            base.OnResume();
+            ActivityMediator.Instance.Cancel();
+        }
+
+        protected override void OnNewIntent(Intent intent)
+        {
+            base.OnNewIntent(intent);
+            ActivityMediator.Instance.Send(intent.DataString);
```
Comment (lbalmaceda) [inline src/Auth0.OidcClient.Android/Auth0ClientActivity.cs]: If this line resolves to parse an authentication result (either good or bad) that is where the call should end, right? No need to pass down the intent with `base.OnNewIntent(intent)`. Unless.. you can't know if the intent received here is a result of an authentication. In that case, keep both lines.

File: src/Auth0.OidcClient.Android/ChromeCustomTabsWebView.cs [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+using Android.Support.CustomTabs;
+
+namespace Auth0.OidcClient
+{
+    public class ChromeCustomTabsWebView : AndroidWebViewBase
+    {
+        protected override void LaunchBrowser(Android.Net.Uri uri)
+        {
+            var builder = new CustomTabsIntent.Builder();
+            var customTabsIntent = builder.Build();
```
Comment (lbalmaceda) [inline src/Auth0.OidcClient.Android/ChromeCustomTabsWebView.cs]: if you're not going to use the `builder` then you can merge both lines above into 1.

```js
var customTabsIntent = new CustomTabsIntent.Builder().Build();
```

File: src/Auth0.OidcClient.Android/Resources/AboutResources.txt [hunk 4]
```diff
@@ -1,44 +0,0 @@
-Images, layout descriptions, binary blobs and string dictionaries can be included
```
Comment (lbalmaceda) [inline src/Auth0.OidcClient.Android/Resources/AboutResources.txt]: Why was this removed?

File: src/Auth0.OidcClient.Android/Resources/Resource.Designer.cs [hunk 5]
```diff
@@ -27,6 +27,33 @@ static Resource()
 		public partial class Attribute
 		{
 			
+			// aapt resource value: 0x7f010007
+			public static int font = 2130771975;
```
Comment (lbalmaceda) [inline src/Auth0.OidcClient.Android/Resources/Resource.Designer.cs]: 🙀If this is generated by you, as the file you removed says, use `R` references instead of doing that. The next release may change these values and break your build. If it's auto-generated, ignore this comment.

File: src/Auth0.OidcClient.Android/SystemBrowserWebView.cs [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+using Android.Content;
+
+namespace Auth0.OidcClient
+{
+    public class SystemBrowserWebView : AndroidWebViewBase
+    {
+        protected override void LaunchBrowser(Android.Net.Uri uri)
+        {
+            var intent = new Intent(Intent.ActionView, uri);
+            intent.AddFlags(ActivityFlags.NoHistory)
```
Comment (lbalmaceda) [inline src/Auth0.OidcClient.Android/SystemBrowserWebView.cs]: Do you offer a way to disable custom tabs? if not, why are you keeping this? By default if the custom tabs intent fails to find a compatible custom-tabs-browser, it will use any browser app that can resolve an ActionView intent.

File: test/Android/Resources/layout/Main.axml [hunk 7]
```diff
@@ -15,7 +15,7 @@
         android:text="@string/Logout" />
     <TextView
         android:id="@+id/UserDetailsTextView"
-        android:text="Text"
+		android:scrollbars="vertical"
```
Comment (lbalmaceda) [inline test/Android/Resources/layout/Main.axml]: is this enough to fix the scrolling issue?

File: src/Auth0.OidcClient.Android/AndroidWebViewBase.cs (hunk 1 above)
Comment (damieng) [reply src/Auth0.OidcClient.Android/AndroidWebViewBase.cs]: It's set by the OIDC IdentityModel internally.

File: src/Auth0.OidcClient.Android/Auth0ClientActivity.cs (hunk 2 above)
Comment (damieng) [reply src/Auth0.OidcClient.Android/Auth0ClientActivity.cs]: Yeah I don't want to bypass any of their own custom onnewintent code.

File: src/Auth0.OidcClient.Android/Resources/AboutResources.txt (hunk 4 above)
Comment (damieng) [reply src/Auth0.OidcClient.Android/Resources/AboutResources.txt]: Was just a boilerplate text file - added nothing to the lib.

File: src/Auth0.OidcClient.Android/Resources/Resource.Designer.cs (hunk 5 above)
Comment (damieng) [reply src/Auth0.OidcClient.Android/Resources/Resource.Designer.cs]: I didn't add this - all the `.Designer.` files in .NET are generated/controlled by the tooling.

File: src/Auth0.OidcClient.Android/SystemBrowserWebView.cs (hunk 6 above)
Comment (damieng) [reply src/Auth0.OidcClient.Android/SystemBrowserWebView.cs]: Wanted to keep it around for a while for backward-compatibility in case people have problems with custom chrome tabs.

File: test/Android/Resources/layout/Main.axml (hunk 7 above)
Comment (damieng) [reply test/Android/Resources/layout/Main.axml]: It is.

File: src/Auth0.OidcClient.Android/AndroidWebViewBase.cs (hunk 1 above)
Comment (damieng) [reply src/Auth0.OidcClient.Android/AndroidWebViewBase.cs]: It's set in the ActivityMediator.

File: src/Auth0.OidcClient.Android/SystemBrowserWebView.cs (hunk 6 above)
Comment (lbalmaceda) [reply src/Auth0.OidcClient.Android/SystemBrowserWebView.cs]: @damieng Ok. But mark it as deprecated at least so they are aware it's going to be removed soon.

Comment (damieng) [reply src/Auth0.OidcClient.Android/SystemBrowserWebView.cs]: Will deprecate it in a couple of months.  (Deprecated attribute is often a compiler error in .NET as people turn on "treat warnings as errors")
